pkg/rtc/transport.go

```diff
@@ -1007,17 +1007,50 @@ func (t *PCTransport) configureReceiverDTXAndStereo(enableDTX bool) {
 		}
 
 		//
-		// As `SetCodecPreferences` on a transceiver replaces all codecs,
-		// cycle through sender codecs also and add them before calling
-		// `SetCodecPreferences`
+		// As peer connection is used in only in send or receive mode,
+		// can modify the relevant direct and set codec preferences.
 		//
-		var senderCodecs []webrtc.RTPCodecParameters
+		err := transceiver.SetCodecPreferences(modifiedReceiverCodecs)
+		if err != nil {
+			t.params.Logger.Warnw("failed to SetCodecPreferences", err)
+		}
+	}
+}
+
+func (t *PCTransport) configureSenderStereo() {
+	transceivers := t.pc.GetTransceivers()
+	for _, transceiver := range transceivers {
+		if transceiver.Kind() != webrtc.RTPCodecTypeAudio {
+			continue
+		}
+
 		sender := transceiver.Sender()
-		if sender != nil {
-			senderCodecs = sender.GetParameters().Codecs
+		if sender == nil {
+			continue
 		}
 
-		err := transceiver.SetCodecPreferences(append(modifiedReceiverCodecs, senderCodecs...))
+		var modifiedSenderCodecs []webrtc.RTPCodecParameters
+
+		senderCodecs := sender.GetParameters().Codecs
+		for _, senderCodec := range senderCodecs {
+			if senderCodec.MimeType == webrtc.MimeTypeOpus {
```

Comment:
is it because of this check? I think MimeType is `audio/red` for Opus RED

Reply:
I just tried changing it, and the line on recieverCodec to be:
`if receiverCodec.MimeType == webrtc.MimeTypeOpus || receiverCodec.MimeType == "audio/red" {`
`if senderCodec.MimeType == webrtc.MimeTypeOpus || senderCodec.MimeType == "audio/red" {`

It didn't seem to make a difference. There's more discussion on #970 as well.